fitbenchmarking/controllers/aoclda_controller.py

```diff
@@ -0,0 +1,327 @@
+        :rtype: int
+        """
+        H, _ = data.cost_func.hes_res(x)
+        Hr[:] = np.matmul(H, r)
+        return 0
+
+    def setup(self):
+        """
+        Setup for AOCLDA
+        """
+
+        n_coef = len(self.initial_params)
+        n_res = len(self.data_y)
+        # self.VERBOSE = 0 # inherit from -e parameter (print level 0..3)
+
+        # Use bytestrings explicitly as python 3 defaults to unicode.
+        if self.minimizer == "gn":
+            model = b"gauss-newton"  # self._options[b"model"] = 1
+            method = b"galahad"  # self._options[b"nlls_method"] = 4
+            glob_strategy = b"tr"  # self._options[b"type_of_method"] = 1
+        elif self.minimizer == "gn_reg":
+            model = b"gauss-newton"  # self._options[b"model"] = 1
+            method = b"galahad"  # self._options[b"nlls_method"] = 4
+            glob_strategy = b"reg"  # self._options[b"type_of_method"] = 2
+        elif self.minimizer == "hybrid":
+            model = b"hybrid"  # self._options[b"model"] = 3
+            method = b"galahad"  # self._options[b"nlls_method"] = 4
+            glob_strategy = b"tr"  # self._options[b"type_of_method"] = 1
+        elif self.minimizer == "hybrid_reg":
+            model = b"hybrid"  # self._options[b"model"] = 3
+            method = b"galahad"  # self._options[b"nlls_method"] = 4
+            glob_strategy = b"reg"  # self._options[b"type_of_method"] = 2
+        elif self.minimizer == "newton":
+            model = b"quasi-newton"  # self._options[b"model"] = 2
+            method = b"galahad"  # self._options[b"nlls_method"] = 4
+            glob_strategy = b"tr"  # self._options[b"type_of_method"] = 1
+        elif self.minimizer == "newton_reg":
+            model = b"quasi-newton"  # self._options[b"model"] = 2
+            method = b"galahad"  # self._options[b"nlls_method"] = 4
+            glob_strategy = b"reg"  # self._options[b"type_of_method"] = 2
+        elif self.minimizer == "newton-tensor":
+            model = b"tensor-newton"  # self._options[b"model"] = 4
+            method = b"galahad"  # self._options[b"nlls_method"] = 4
+            glob_strategy = b"tr"  # self._options[b"type_of_method"] = 1
+        elif self.minimizer == "newton-tensor_reg":
+            model = b"tensor-newton"  # self._options[b"model"] = 4
+            method = b"galahad"  # self._options[b"nlls_method"] = 4
+            glob_strategy = b"reg"  # self._options[b"type_of_method"] = 2
+        else:
+            raise UnknownMinimizerError(
+                f"No {self.minimizer} minimizer for AOCLDA"
+            )
+
+        # If parameter ranges have been set in problem, then set up bounds
+        # option. For AOCLDA, this must be a 2 tuple array like object,
+        # the first tuple containing the lower bounds for each parameter
+        # and the second containing all upper bounds.
+        if self.value_ranges is not None:
+            self._lb, self._ub = zip(*self.value_ranges)
```

Comment:
I tried to run a problem with bounds (e.g. the CrystalField problems) and got an error about incompatible argument types

Reply:
Apologies for the long delay in responding to this - FitBenchmarking only accepts SIF files in a very specific format so the issue here is that the SIF file doesn't contain any data values. We do have some other example problems with parameter bounds in the [SIF_GO](https://github.com/fitbenchmarking/fitbenchmarking/tree/master/examples/benchmark_problems/SIF_GO) directory.